File: gateway-webapp/src/app/models/Note.ts
export class Note {
  id: number;
  title: string;
  text: string;

  constructor(id: number, title: string, text: string) {
    this.id = id;
    this.title = title;
    this.text = text;
  }
}

export const NULL_NOTE: Note = new Note(0, '', '');

File: gateway-api/src/main/kotlin/absaliks/gateway/notes/Note.kt
package absaliks.gateway.notes

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class Note(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int,
    val title: String,
    val text: String
)